eng/scripts/dependencies/generate-deps.ps1

```diff
@@ -0,0 +1,36 @@
+param(
+  [string]$PackagesPath,
+  [string]$DepsOutputFile,
+  [string]$ProjectRefPath
+)
+
+#-Packages (dir -include "$(System.ArtifactsDirectory)/BuildArtifacts/Azure.*.nupkg" -exclude *symbols.nupkg)
+#-DepsOutputFile "$(Build.ArtifactStagingDirectory)/reports/azure-sdk.deps.json"
+
+mkdir $ProjectRefPath -force | out-null
+$projRefsPath = Join-Path $ProjectRefPath "azure-sdk-projectreferences.props"
+
+$projectRefsContents = @"
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup>
+
+"@
+
+foreach ($pkg in (Get-ChildItem "$PackagesPath/*" -Include *.nupkg -Exclude *.symbols.nupkg))
+{
+  if ($pkg.Name -match "(?<pkg>[^\\]+)\.(?<ver>\d+\.\d+\.\d+.*)\.nupkg") 
+  {
+    $projectRefsContents += "    <PackageReference Include=`"$($matches['pkg'])`" Version=`"$($matches['ver'])`" />`n"
+  }
+}
+
+$projectRefsContents += @"
+  </ItemGroup>
+</Project>
+"@
+
+Set-Content -Path $projRefsPath -Value $projectRefsContents
+
+Write-Host "dotnet build $PSScriptRoot/azure-sdk-deps.csproj /t:GenerateDepsFile /p:PublishDepsFilePath=$DepsOutputFile /p:AzureSdkProjectReferencesPath=$projRefsPath /p:RestoreAdditionalProjectSources=$PackagesPath"
```

Comment:
Rather than keep these in sync, just assign the command line to a variable, write-host it, and `iex` it, e.g.:

```powershell
$cmd = "dotnet build ..."
write-host $cmd
invoke-expression $cmd
```